easy-rules-jexl/src/test/java/org/jeasy/rules/jexl/JexlActionTest.java

```diff
@@ -0,0 +1,219 @@
+import org.jeasy.rules.api.Action;
+import org.jeasy.rules.api.Facts;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.rules.ExpectedException;
+
+public class JexlActionTest {
+
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule().enableLog();
+
+    @SuppressWarnings("deprecation")
+    @Rule
+    public final ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void testMVELActionExecution() throws Exception {
+        // given
+        Action markAsAdult = new JexlAction("person.setAdult(true);");
+        Facts facts = new Facts();
+        Person foo = new Person("foo", 20);
+        facts.put("person", foo);
+
+        // when
+        markAsAdult.execute(facts);
+
+        // then
+        assertThat(foo.isAdult()).isTrue();
+    }
+
+    @Test
+    public void testJexlFunctionExecution() throws Exception {
+        // given
+        Action printAction = new JexlAction("var hello = function() { System.out.println(\"Hello from JEXL!\"); }; hello();");
+        Facts facts = new Facts();
+        facts.put("System", System.class);
+
+        // when
+        printAction.execute(facts);
+
+        // then
+        assertThat(systemOutRule.getLog()).contains("Hello from JEXL!");
+    }
+
+    @Test
+    public void testMVELActionExecutionWithFailure() throws Exception {
+        // given
+        expectedException.expect(JexlException.Method.class);
+        expectedException.expectMessage("org.jeasy.rules.jexl.JexlAction.<init>@1:7 unsolvable function/method 'setBlah'");
+        Action action = new JexlAction("person.setBlah(true);");
+        Facts facts = new Facts();
+        Person foo = new Person("foo", 20);
+        facts.put("person", foo);
+
+        // when
+        action.execute(facts);
+
+        // then
+        // excepted exception
+    }
+
+    @Test
+    public void testJexlActionWithExpressionAndFacts() throws Exception {
+        // given
+        JexlEngine jexl = new JexlBuilder()
+                .create();
+        Action printAction = new JexlAction(
+                "var random = function() { System.out.println(\"Random from JEXL = \" + new('java.util.Random', 123).nextInt(10)); }; random();",
+                jexl);
+        Facts facts = new Facts();
+        facts.put("System", System.class);
+
+        // when
+        printAction.execute(facts);
+
+        // then
+        assertThat(systemOutRule.getLog()).contains("Random from JEXL = 2");
+    }
+
+    @Test
+    public void testWithBlackSanbox() throws Exception {
+        JexlSandbox sandbox = new JexlSandbox(false);
+        sandbox.white(System.class.getName()).execute("currentTimeMillis");
+        JexlEngine jexl = new JexlBuilder()
+                .sandbox(sandbox)
+                .create();
+
+        Facts facts = new Facts();
+        facts.put("result", null);
```

Comment:
This is not type safe. Even though it is possible, I would not use it in real case scenario. While defining a method in an EL expression and call it later on is ok, assigning a variable inside the expression and get the result outside is not a good practice IMO. This comment is about the practice itself rather than the code (again, as mentioned in my other comment in this PR, my concern is that people will just do the same). Each time I write something like this, there is a little voice in my head reminding me that the end goal is not to put code in Strings 😄